### ISSUE TITLE
LoadWith

### ISSUE THREAD

**Issue by genusP** [linq2db/linq2db#832]
Are there any plans to work on LoadWith to support the enumerable properties?

**Comment by ili** [comment]
In general yes, we do have an idea that LoadWith should be totally redone to minify database queries & support more cases.

It is really huge task, so I cannot say for sure when...

**Comment by genusP** [comment]
IMHO this is quite an important feature and it's strange that it is not supported.

You could specify the release in which these changes are expected.

There are some ways to circumvent this limitation. I use linq2db as a data access layer  in OData WebApi. OData builds expressions, for example:
```C#
db.GetTable<Entity>().Select(p=> new SelectAllAndExpand{
                    ModelID = "2cd45fee-a6aa-4d0e-9024-0a7fbc48e50b",
                    Instance = Param_0,
                    Container = new NamedProperty()
                    {
                        Name = "EnumerableProperty",
                        Value = Param_0.EnumerableProperty.Select(p2=>new SelectAll
                                    {
                                        ModelID = "2cd45fee-a6aa-4d0e-9024-0a7fbc48e50b",
                                        Instance = Param_1
                                    })
                   }
})
```

**Comment by MaceWindu** [comment]
I think support of IEnumerable is not related to LoadWith optimization and could be done easily

**Comment by sdanyliv** [comment]
It is related to LoadWith, such support should be handled by LoadWith functionality.

**Comment by MaceWindu** [comment]
@genusP could you provide more details what do you mean by enumerable properties support?

**Comment by genusP** [comment]
@MaceWindu, if I use association property (returning IEnumerable), then raised exception with message: Multiple queries are not allowed.

I can not enable AllowMultipleQuery, because there can be many requested objects to them, respectively, requests.

**Comment by genusP** [comment]
Any plans for this feature

**Comment by sdanyliv** [comment]
Thinking about this from time to time. It needs many efforts to implement. And possibly huge @igor-tkachev help.

**Comment by genusP** [comment]
I understand. Thanks for the reply.

**Comment by genusP** [comment]
Yours are plan to fix this this issue in new version 3.0?

**Comment by genusP** [comment]
#1756
